### PR TITLE
components: Add theme provider

### PR DESCRIPTION
**Pull request by siddharthkp**
Merge after https://github.com/codesandbox/codesandbox-client/pull/3292

Moved all the logic of `./storybook/theme` to a `<ThemeProvider>`. This serves as a boundary to a part of the interface which is getting redesigned. Eventually our entire application can be wrapped in just 1 of these.

This also gives us the chance to apply _globals_ inside the boundary.

### CODE REVIEW

**Comment by lbogdan** [comment]
Build for latest commit https://github.com/codesandbox/codesandbox-client/pull/3294/commits/c1dcb080e0a575ac65ade647f0f3bdf919eb6e15 is at https://pr3294.build.csb.dev/s/new.

File: packages/components/.storybook/config.tsx [hunk 1]
```diff
@@ -34,8 +33,10 @@ const GlobalStyle = createGlobalStyle`
   ${global};
   body {
     font-family: 'Inter', sans-serif;
-    background-color: ${theme.colors.sideBar.background};
-    color: ${theme.colors.sideBar.foreground};
+    margin: 0;
+    background-color: ${props =>
+      props.theme.colors.sideBar.background} !important;
+    color: ${props => props.theme.colors.sideBar.foreground} !important;
```
Comment (SaraVieira) [inline packages/components/.storybook/config.tsx]: its missing the border-box thing

```
const GlobalStyle = createGlobalStyle`
  ${global};
  html body {
    font-family: 'Inter', sans-serif;
    background-color: ${theme.colors.sideBar.background};
    color: ${theme.colors.sideBar.foreground};
    margin: 0;
    padding: 20px;

    * {
      box-sizing: border-box;
    }
  }
`;
```

Comment (siddharthkp) [reply packages/components/.storybook/config.tsx]: ah yes, will merge with master and bring this along

Comment (siddharthkp) [reply packages/components/.storybook/config.tsx]: added box-sizing, removed padding though. Do we need it on body?

**Pull request merged**